src/templates/cmpapi.js

```diff
@@ -119,19 +119,19 @@
   }
 
   var channelId = '<%-CHANNEL_ID%>';
-
+  var cmpEnabled = '<%-CMP_ENABLED%>' === 'true';
   var outOfSample = Math.floor(Math.random() * 100) + 1 > parseInt('<%-CMP_ENABLED_SAMPLING_THRESHOLD_PERCENT%>');
+  var now = Date.now();
 
   var technicalCookie = parseInt(readStorageOrCookie('<%-TECH_COOKIE_NAME%>'));
   if (!technicalCookie) {
-    technicalCookie = Date.now();
-    writeStorageAndCookie('<%-TECH_COOKIE_NAME%>', technicalCookie);
+    technicalCookie = now;
+    if (cmpEnabled) {
```

Comment:
Add this check to the `writeStorageAndCookie` so that we can assure everywhere that localstorage is not touched if CMP is disabled